dev/manual_tests/lib/menu_bar.dart

```diff
@@ -0,0 +1,317 @@
+  'Menu 3',
+  'Menu 4',
+];
+
+const List<String> subMenu = <String>[
+  'Sub Menu 1',
+  'Sub Menu 2',
+  'Sub Menu 3',
+  'Sub Menu 4',
+  'Sub Menu 5',
+  'Sub Menu 6',
+  'Sub Menu 7',
+];
+
+const List<String> subSubMenu = <String>[
+  'Sub Sub Menu 1',
+  'Sub Sub Menu 2',
+];
+
+void main() {
+  debugFocusChanges = false;
+  runApp(const MaterialApp(
+    title: 'Menu Tester',
+    home: Material(child: Home()),
+  ));
+}
+
+class Home extends StatefulWidget {
+  const Home({Key? key}) : super(key: key);
+
+  @override
+  State<Home> createState() => _HomeState();
+}
+
+class _HomeState extends State<Home> {
+  late MenuBarController controller;
+  bool isPlatformMenu = false;
+  VisualDensity density = VisualDensity.standard;
+  TextDirection textDirection = TextDirection.ltr;
+  bool enabled = true;
+  bool checked = false;
+
+  void _activateItem(String item) {
+    debugPrint('Activated item $item');
+  }
+
+  void _openItem(String item) {
+    debugPrint('Opened item $item');
+  }
+
+  void _closeItem(String item) {
+    debugPrint('Closed item $item');
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    controller = MenuBarController();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    return Directionality(
+      textDirection: textDirection,
+      child: Builder(builder: (BuildContext context) {
+        return Theme(
+          data: theme.copyWith(visualDensity: density),
+          child: MenuBar.adaptive(
+            targetPlatform: isPlatformMenu ? TargetPlatform.macOS : TargetPlatform.windows,
+            enabled: enabled,
+            controller: controller,
+            body: Center(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget>[
+                  Row(
+                    mainAxisSize: MainAxisSize.min,
+                    children: <Widget>[
+                      Checkbox(
+                        value: isPlatformMenu,
+                        onChanged: (bool? value) {
+                          setState(() {
+                            if (value ?? false) {
+                              isPlatformMenu = true;
+                            } else {
+                              isPlatformMenu = false;
+                            }
+                          });
+                        },
+                      ),
+                      const Text('Platform Menus')
+                    ],
+                  ),
+                  ConstrainedBox(
+                    constraints: const BoxConstraints.tightFor(width: 400),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.end,
+                      children: <Widget>[
+                        Text('Horizontal Density: ${density.horizontal.toStringAsFixed(1)}'),
+                        Slider(
+                            value: density.horizontal,
```

Comment:
Indentation in this example varies between 4 and 2

Reply:
Thanks.  This is still a draft though. I'll do a formatting pass before I send it to review.